pyproject.toml

```diff
@@ -97,6 +97,8 @@ select = [
     "NPY002",
     # Ruff-specific rules
     "RUF",
+    # banned-api
+    "TID251",
```

Comment:
This is handy! Could we use it to replace https://github.com/rapidsai/cudf/blob/branch-25.04/python/cudf/cudf/utils/utils.py#L132? We'd have to add more noqas where that tool will automatically allowlist the places where we want to use these functions, but it could still be worth it to simplify the code and go with a static analysis rather than a runtime decorator and stack tracing approach. It might also speed up our test suite to not have that running.

Reply:
I see this decorator is currently used on class properties (`MulitiIndex.codes` and `DataFrame.columns`), while this ruff rule _seems_ to work on importable objects. I'll experiment to see this Ruff rule can supersede `_external_only_api`